### PR TITLE
NUC-2025 / don't go negative

### PR DESCRIPTION
**Pull request by gjesse**

### CODE REVIEW

**Comment by gjesse** [comment]
i can explain this a bit more in person if needed

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java [hunk 1]
```diff
@@ -284,8 +288,14 @@ int deleteJobFromReady(
             "     redis.call('SREM', $multiChannelSet$, $channelPrefix$)\n" +
             "  end\n" +
             "end\n" +
-            "if removed > 0 then\n" +
-            "  redis.call('DECRBY', $runningCount$, removed)\n" +
```
Comment (amanduggal) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java]: ah.. we weren't looking at the removed from running queue first.

File: rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java [hunk 2]
```diff
@@ -284,8 +288,14 @@ int deleteJobFromReady(
...
+            "if removedFromRunning > 0 then\n" +
```
Comment (amanduggal) [inline rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java]: do you need to do something like 
` if removedFromRunning and tonumber(removedFromRunning)` here as well?

Comment (gjesse) [reply rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelSchedulerDAO.java]: no, the zrem call should always return the number of entries removed, where as the if thing and tonumber(thing) is needed for keys that may not exist

**Pull request merged**